File: apps/fogg/src/pump/impl.go
package pump

import (
	"fogg/src/pump/want"
	"fogg/src/type"
)

func Impl(address types.Address, payload types.Payload, options types.Options) int {
	if want.None(options) {
		return none(address, payload, options)
	}
	if want.Once(options) {
		return once(address, payload, options)
	}
	if want.Many(options) {
		return many(address, payload, options)
	}
	return -1
}

File: apps/fogr/src/pump/impl.rs
use crate::pump::many::many;
use crate::pump::none::none;
use crate::pump::once::once;
use crate::pump::want;
use crate::r#type::{Address, Options, Payload};

pub fn r#impl(address: Address, payload: Payload, options: Options) -> i32 {
    if want::none(options) {
        return none(address, payload, options);
    }
    if want::once(options) {
        return once(address, payload, options);
    }
    if want::many(options) {
        return many(address, payload, options);
    }
    -1
}
